Sources/FoundationEssentials/FileManager/FileManager+Files.swift

```diff
@@ -553,49 +553,61 @@ extension _FileManagerImpl {
     func attributesOfItem(atPath path: String) throws -> [FileAttributeKey : Any] {
 #if os(Windows)
         return try path.withNTPathRepresentation { pwszPath in
-            var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
-            guard GetFileAttributesExW(pwszPath, GetFileExInfoStandard, &faAttributes) else {
-                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
-            }
-
             let hFile = CreateFileW(pwszPath, GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, nil, OPEN_EXISTING, 0, nil)
             if hFile == INVALID_HANDLE_VALUE {
                 throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
             }
             defer { CloseHandle(hFile) }
 
+            var info: BY_HANDLE_FILE_INFORMATION = BY_HANDLE_FILE_INFORMATION()
+            GetFileInformationByHandle(hFile, &info)
+
             let dwFileType = GetFileType(hFile)
             let fatType: FileAttributeType = switch (dwFileType) {
                 case FILE_TYPE_CHAR: FileAttributeType.typeCharacterSpecial
                 case FILE_TYPE_DISK:
-                    faAttributes.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY
+                    info.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY
                             ? FileAttributeType.typeDirectory
                             : FileAttributeType.typeRegular
                 case FILE_TYPE_PIPE: FileAttributeType.typeSocket
                 case FILE_TYPE_UNKNOWN: FileAttributeType.typeUnknown
                 default: FileAttributeType.typeUnknown
             }
 
-            let size: UInt64 = (UInt64(faAttributes.nFileSizeHigh) << 32) | UInt64(faAttributes.nFileSizeLow)
-            let creation: Date = Date(timeIntervalSince1970: faAttributes.ftCreationTime.timeIntervalSince1970)
-            let modification: Date = Date(timeIntervalSince1970: faAttributes.ftLastWriteTime.timeIntervalSince1970)
-            return [
+            let systemNumber = UInt64(info.dwVolumeSerialNumber)
+            let systemFileNumber = UInt64(info.nFileIndexHigh << 32) | UInt64(info.nFileIndexLow)
+            let referenceCount = UInt64(info.nNumberOfLinks)
+
+            let isReadOnly = info.dwFileAttributes & FILE_ATTRIBUTE_READONLY != 0
```

Comment:
I think that it is safer to ensure that the bit is set ...

```suggestion
            let isReadOnly = info.dwFileAttributes & FILE_ATTRIBUTE_READONLY == FILE_ATTRIBUTE_READONLY
```

Reply:
Personally I prefer the `!= 0` version. If we're doing a bitwise AND, we _know_ the answer is either `0` or `FILE_ATTRIBUTE_READONLY`. The only way we could trip up is if `FILE_ATTRIBUTE_READONLY` is not really a bit mask, but that would be wrong for _either_ formulation (and we already know it's a bit mask anyway).